### PR TITLE
Bug 1859744 - After installing an add-on add the metadata from installed extension.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1859744

### CODE REVIEW

File: android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt [hunk 1]
```diff
@@ -134,8 +134,8 @@ class AddonManager(
             id = addon.id,
             url = addon.downloadUrl,
             onSuccess = { ext ->
-                val installedAddon = addon.copy(installedState = ext.toInstalledState())
-                addonUpdater.registerForFutureUpdates(installedAddon.id)
+                val installedAddon = Addon.newFromWebExtension(ext, ext.toInstalledState())
```
Comment (willdurand) [inline android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt:137]: We don't need to pass `ext.toInstalledState()` anymore, right?

**Review by willdurand — APPROVED** [review]
r+wc, thanks!

File: android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt (hunk 1 above)
Comment (Amejia481) [reply android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt:137]: We haven't land that [change](https://github.com/mozilla-mobile/firefox-android/blob/a25185f9967e544213f2333b1a1d45591940e7bd/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/Addon.kt#L293) yet 😅

Comment (willdurand) [reply android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt:137]: hahaha

**Pull request merged**